### ISSUE TITLE
Atuin with mosh won't repaint properly

### ISSUE THREAD

**Issue by jaysonsantos** [atuinsh/atuin#1324]
First, thank you very much for writing atuin, it is really useful, and it is allowing me to slowly shift away from fzf setup.

**Describe the bug**
When using atuin on a remote (or even localhost for tests) connected with mosh, it will not repaint properly.

**To Reproduce**
1. mosh remote -server/mosh localhost
2. ls -al (so you fill the whole screen with data)
3. ctrl-r should be broken

**Expected behavior**
It should repaint the same way as running locally

**Device (please complete the following information):**
 - OS: [e.g. macOS] macos
 - Shell [bash/zsh/fish/nu] zsh
 - Version [e.g. 16.0.0] zsh 5.9 (arm-apple-darwin21.3.0)

**Additional context**
Here is what happens visually
![image](https://github.com/atuinsh/atuin/assets/51510/f7c57569-e528-443e-8399-dbca443c066f)

**Comment by conradludgate** [comment]
Thanks for the report. It likely seems to be something going wrong with ratatui/crossterm

**Comment by jaysonsantos** [comment]
Hey @conradludgate i will relay over

**Comment by dcarosone** [comment]
This is the same behaviour that happens in screen, unless the terminal 'altscreen' is enabled.

I see it with mosh too, now that I check. I hadn't noticed because when I use mosh there's normally a screen running at the remote end, and having screen provide the alt mode then hides the problem again.

atuin / ratatui wants to put the terminal into alternate screen mode, so that the original screen contents can be restored when the text overlay has been removed.  Something in mosh is preventing that.  Literally issue 2 in the mosh repo is about it, and it's a very long discussion.

**Comment by dcarosone** [comment]
This seems to be the definitive answer / current state https://github.com/mobile-shell/mosh/issues/843

**Comment by jaysonsantos** [comment]
@dcarosone a solution that worked was openning mosh, then zellij so atuin can be used

**Comment by dcarosone** [comment]
> a solution that worked was openning mosh, then zellij so atuin can be used

Yep, that's the same workaround as screen.

The issue that remains in atuin is not specific to mosh, and is about doing something sensible when there's no alt-screen mode available in the terminal we're given. From a quick glance at ratatui and crossterm docs, there are `EnterAlternateScreen` and `LeaveAlternateScreen` methods, but I don't see a way to determine whether the hosting terminal has that capability.

The simplest workaround is probably to send a clear screen command right after the switch to the alternate screen, and right before leaving. If we switched, this will have no effect. If the switch did nothing, then the main screen will be cleared and at least prevent garbled content.  This appears to be what vi (nvi at least) does on such terminals.

I'll go looking to see where this happens in atuin a bit later.

**Comment by jaysonsantos** [comment]
But, sending clear screen wouldn't make the user lose the previous contents before atuin has painted?

**Comment by dcarosone** [comment]
Yes, it would. There's no way to avoid this without the alternate screen mode (that's, basically, what it's there for).

**Comment by DwordPtr** [comment]
I'm affected too.

**Comment by dcarosone** [comment]
as of the latest release, the defaults now don't use the alternate/full screen mode, so this should be less intrusive

**Comment by jyn514** [comment]
i was looking for the reverse, to *force* atuin to use the alternate screen. it turns out the current way to do that is to set `inline_height = 0` (which apparently means 'full-screen').

**Comment by cameronbraid** [comment]
FYI I had a similar issue on KConsole where after atuin exited (after selecting a command form histroy) the screen wasn't restored properly always leaving my prompt at the top of the console.

 Setting `inline_height = 0` fixes this for me
